### PR TITLE
[Breadcrumbs]: Documentation, aria-label construction, responsiveness, refactoring

### PR DESCRIPTION
**Pull request by RiinaKuu**
https://jira.funidata.fi/browse/DS-145

- Breadcrumbs refactoring
- IdService additions, adjustments & unit tests
- Implementing IdService to components
- Adjusting components tests

### CODE REVIEW

File: ngx-fudis/projects/ngx-fudis/src/lib/components/breadcrumbs/breadcrumbs.component.ts [hunk 1]
```diff
@@ -8,7 +9,24 @@ import { FudisBreadcrumb } from '../../types/miscellaneous';
...
+	constructor(private _translationService: FudisTranslationService) {
+		effect(() => {
+			this._breadcrumbsPrefix = this._translationService.getTranslations()().BREADCRUMBS.PREFIX;
+		});
+	}
 
+	/**
+	 * Label to attach to aria-label
+	 */
+	@Input({ required: true }) breadcrumbsLabel: string;
```
Comment (videoeero) [inline ngx-fudis/projects/ngx-fudis/src/lib/components/breadcrumbs/breadcrumbs.component.ts:21]: Should this be just 'label'?

File: ngx-fudis/projects/ngx-fudis/src/lib/components/breadcrumbs/breadcrumbs.component.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<nav [attr.aria-label]="breadcrumbsAriaLabel" *ngIf="links.length > 0">
+<nav [attr.aria-label]="_breadcrumbsPrefix + ': ' + breadcrumbsLabel" *ngIf="links.length > 0">
 	<ul class="fudis-breadcrumbs">
 		<li class="fudis-breadcrumbs__item" *ngFor="let link of links; let last = last">
 			<fudis-link
```
Comment (videoeero) [inline ngx-fudis/projects/ngx-fudis/src/lib/components/breadcrumbs/breadcrumbs.component.html:4]: Should we have 'aria-current' attribute set to the last link in the breadcrumbs?

File: ngx-fudis/projects/ngx-fudis/src/lib/components/breadcrumbs/breadcrumbs.component.ts (hunk 1 above)
Comment (RiinaKuu) [reply ngx-fudis/projects/ngx-fudis/src/lib/components/breadcrumbs/breadcrumbs.component.ts:21]: Yes, good call for uniformity.

File: ngx-fudis/projects/ngx-fudis/src/lib/components/breadcrumbs/breadcrumbs.component.html (hunk 2 above)
Comment (RiinaKuu) [reply ngx-fudis/projects/ngx-fudis/src/lib/components/breadcrumbs/breadcrumbs.component.html:4]: There is aria-current in fudis-link element and it works like this at the moment. Do we need something else?

Screenshot 2023-10-26 at 9 22 43

File: ngx-fudis/projects/dev/src/app/app.component.ts [hunk 3]
```diff
@@ -69,6 +69,16 @@ export class AppComponent implements OnInit {
 		};
 	});
 
+	breadcrumbsLinks = [
+		{ label: 'Fudis', url: '/example' },
+		{ label: 'Sandbox', url: '/example' },
```
Comment (videoeero) [inline ngx-fudis/projects/dev/src/app/app.component.ts:74]: I think we should create this Sandbox example using Transloco.

File: ngx-fudis/projects/ngx-fudis/src/lib/types/miscellaneous.ts [hunk 4]
```diff
@@ -56,6 +56,10 @@ export type FudisTextAlign = 'left' | 'right' | 'center';
 export type FudisLanguageAbbr = 'fi' | 'sv' | 'en';
 
 export interface FudisTranslationConfig {
+	BREADCRUMBS: {
+		// Universal prefix to notify user about breadcrumbs navigation
```
Comment (videoeero) [inline ngx-fudis/projects/ngx-fudis/src/lib/types/miscellaneous.ts:60]: 'to notify screenreader user about breadcrumbs navigation' ?

**Pull request merged**